### PR TITLE
Added support in DBUtil class for creating tables with indexes and foreign keys.

### PR DESCRIPTION
**Pull request by calvinfroedge**
Tried to stay as true to Dan's form as possible.  Added some comments before functions where they were missing.  

I could not get return \implode to work for me as in other code.  Had to do return implode.

Tested and all works.  Was successful with my own migration using this code.

### CODE REVIEW

File: classes/dbutil.php [hunk 1]
```diff
@@ -86,6 +99,16 @@ class DBUtil {
 			$primary_keys = \DB::quote_identifier($primary_keys);
 			$sql .= ",\n\tPRIMARY KEY ".$key_name." (" . implode(', ', $primary_keys) . ")";
 		}
+		if( ! empty($indexes))
+		{
+			$sql .= ",";
+			$sql .= static::process_indexes($indexes);
+		}
+		if( ! empty($foreign_keys))
```
Comment (philsturgeon) [inline classes/dbutil.php:107]: This is close but a little off. It should be `if ( ! empty` to keep in line with the style guide.

**Comment by philsturgeon** [comment]
Could you update your branch as this cannot be merged right now. Also, could you make sure it does not cross over with https://github.com/fuel/core/pull/557 as both these changes look good.

File: classes/dbutil.php [hunk 2]
```diff
@@ -186,6 +219,51 @@ class DBUtil {
...
+	 * @return   string    the formated indexes string
+	 */	
+	public static function process_indexes($indexes)
+	{
+		$sql = "";
+		$index_list = array();
+		foreach($indexes as $index)
+		{
+			$index = array_change_key_case($index, CASE_UPPER);
+			($index['UNIQUE'] === true) ? $unique = 'UNIQUE ' : $unique = '';
```
Comment (ushis) [inline classes/dbutil.php:234]: Isn't it better to write:

``` php
<?php
$unique = ($index['UNIQUE'] === true) ? 'UNIQUE ' : '';
```

File: classes/dbutil.php [hunk 3]
```diff
@@ -186,6 +219,51 @@ class DBUtil {
...
+			$sql = "\n\t".$unique."INDEX ".\DB::quote_identifier($index['NAME'])." (" . \DB::quote_identifier($index['FIELD']) . " " . $index['ORDER']. ")";
+			$index_list[] .= $sql;
```
Comment (ushis) [inline classes/dbutil.php:236]: I think it should be

``` php
<?php
$index_list[] = $sql;
```

without the dot.

**Comment by ushis** [comment]
Damn this pull request is super old. It would be nice to have this in 1.1

File: classes/dbutil.php (hunk 2 above)
Comment (philsturgeon) [inline classes/dbutil.php:234]: Agreed, but its pretty minor.

**Comment by billmn** [comment]
agree

**Pull request closed without merging**